README.md

```diff
@@ -57,89 +57,113 @@ export default {
+Note that the config option you provide will be passed to the `ESLint` class.
+This is a different set of options than what you'd specify in `package.json` or `.eslintrc`.
+See the [eslint docs](https://eslint.org/docs/developer-guide/nodejs-api#%E2%97%86-new-eslint-options) for more details.
 
-### `cache`
+**Warning**: In eslint-webpack-plugin version 1 the options were passed to the now deprecated [CLIEngine](https://eslint.org/docs/developer-guide/nodejs-api#cliengine).
 
-- Type: `Boolean|String`
-- Default: `false`
+### `cache``
+
+- Type: `Boolean`
+- Default: `true`
 
-This option will enable caching of the linting results into a file. This is particularly useful in reducing linting time when doing a full build.
+**Note**: The cache is enabled by default to decrease execution time.
 
-This can either be a `boolean` value or the cache directory path(ex: `'./.eslint-loader-cache'`).
+### `context`
 
-If `cache: true` is used, the cache is written to the `./node_modules/.cache/eslint-loader` directory. This is the recommended usage.
+- Type: `String`
+- Default: `compiler.context`
+
+A string indicating the root of your files.
 
 ### `eslintPath`
 
 - Type: `String`
 - Default: `eslint`
 
-Path to `eslint` instance that will be used for linting. If the `eslintPath` is a folder like a official eslint, or specify a `formatter` option. Now you dont have to install `eslint`.
+Path to `eslint` instance that will be used for linting. If the `eslintPath` is a folder like a official eslint, or specify a `formatter` option. now you dont have to install `eslint`.
+
+### `files`
+
+- Type: `String|Array[String]`
+- Default: `'.'`
+
+Specify directories, files, or globs. Must be relative to `options.context`.
+Directories are traveresed recursively looking for files matching `options.extensions`.
+File and glob patterns ignore `options.extensions`.
 
 ### `extensions`
 
-- Type: `Array[String]`
-- Default: `['ts', 'js', 'vue']`
+- Type: `String|Array[String]`
+- Default: `'js'`
 
-Extensions that will be used by the loader.
+Specify extensions that should be checked.
 
 ### `fix`
 
 - Type: `Boolean`
 - Default: `false`
 
-This option will enable [ESLint autofix feature](http://eslint.org/docs/user-guide/command-line-interface#fix).
+Will enable [ESLint autofix feature](https://eslint.org/docs/developer-guide/nodejs-api#%E2%97%86-eslint-outputfixes-results).
 
 **Be careful: this option will change source files.**
 
 ### `formatter`
 
 - Type: `String|Function`
-- Default: `stylish`
+- Default: `'stylish'`
+
+Accepts a function that will have one argument: an array of eslint messages (object). The function must return the output as a string. You can use official [eslint formatters](https://eslint.org/docs/user-guide/formatters/).
+
+### `lintDirtyModulesOnly`
+
+- Type: `Boolean`
+- Default: `false`
```

Comment:
What if we enable this by default to speedup initial build and avoid unnecessary warnings that are not related to one's changes?

Reply:
Sure. Enabled 😄